src/services/subscription.service.js

```diff
@@ -0,0 +1,83 @@
+const { loggers } = require('@asymmetrik/node-fhir-server-core');
+const { v4: uuidv4 } = require('uuid');
+const db = require('../storage/DataAccess');
+const topiclist = require('../../public/topiclist.json');
+
+const logger = loggers.get('default');
+const SUBSCRIPTION = 'subscriptions';
+
+module.exports.topiclist = (_args, _context) => {
+  logger.info('Running Subsription $topic-list operation');
+  return new Promise((resolve, _reject) => {
+    resolve(topiclist);
+  });
+};
+
+module.exports.search = (_args, _context) => {
+  return new Promise((resolve, _reject) => {
+    logger.info('Subscription >>> search');
+    resolve(db.select(SUBSCRIPTION, () => true));
+  });
+};
+
+module.exports.searchById = (args, _context) => {
+  return new Promise((resolve, reject) => {
+    let { id } = args;
+    logger.info(`Subscription >>> searchById(${id})`);
+    const result = db.select(SUBSCRIPTION, (r) => r.id === id);
+    if (result.length >= 1) resolve(result);
+    else reject({ message: `Subscription/${id} does not exist` });
+  });
+};
+
+module.exports.create = (_args, { req }) => {
+  return new Promise((resolve, reject) => {
+    const resource = req.body;
+    logger.info(`Subscription >>> create(${resource.id})`);
+    if (!resource) reject({ message: 'Request must contain body.' });
```

Comment:
`reject` doesn't return automatically so hitting some of these with invalid requests means they still get added to the DB